models/Element_OphTrOperationnote_Biometry.php

```diff
@@ -0,0 +1,133 @@
+ * @property Event $event
+ */
+class Element_OphTrOperationnote_Biometry extends Element_OpNote
+{
+	// these are legacy and should be removed one switch to using the constants on the Eye model
+	const LEFT = Eye::LEFT;
+	const RIGHT = Eye::RIGHT;
+	const BOTH = Eye::BOTH;
+
+	private $event;
+
+	/**
+	 * @return bool
+	 */
+	public function hasLeft()
+	{
+		return $this->eye_id != Eye::RIGHT;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function hasRight()
+	{
+		return $this->eye_id != Eye::LEFT;
+	}
+
+	/**
+	 * Returns the static model of the specified AR class.
+	 * @return ElementOperation the static model class
+	 */
+	public static function model($className = __CLASS__)
+	{
+		return parent::model($className);
+	}
+
+	/**
+	 * @return string the associated database table name
+	 */
+	public function tableName()
+	{
+		return 'et_ophtroperationnote_biometry';
+	}
+
+	/**
+	 * @return array customized attribute labels (name=>label)
+	 */
+	public function attributeLabels()
+	{
+		return array(
+			'id' => 'ID',
+			'lens_right' => 'Lens',
+			'lens_left' => 'Lens',
+			'lens_id_right' => 'Lens',
+			'lens_id_left' => 'Lens',
+			'lens_description_left' => 'Description',
+			'lens_description_right' => 'Description',
+			'lens_acon_left' => 'A constant',
+			'lens_acon_right' => 'A constant',
+			'k1_left' => 'K1 (D)',
+			'k1_right' => 'K1 (D)',
+			'k2_left' => 'K2 (D)',
+			'k2_right' => 'K2 (D)',
+			'axis_k1_left' => 'Axis K1 (D)',
+			'axis_k1_right' => 'Axis K1 (D)',
+			'axial_length_left' => 'Axial length (mm)',
+			'axial_length_right' => 'Axial length (mm)',
+			'snr_left' => 'SNR',
+			'snr_right' => 'SNR',
+			'iol_power_left' => 'IOL Power',
+			'predicted_refraction_left' => 'Predicted Refraction',
+			'iol_power_right' => 'IOL Power',
+			'predicted_refraction_right' => 'Predicted Refraction',
+			'target_refraction_left' => 'Target Refraction',
+			'target_refraction_right' => 'Target Refraction',
+		);
+	}
+
+	public function findAll($attributes, $values){
+		// because we are working with a view, we should present the event as the last Biometry from the view
+		// we need the patient ID and the last_modified date of the current event
+		// $attributes == "event_id = ?" in this case
+		$eventData = Event::model()->findByPk($values[0]);
+		$episodeData = Episode::model()->findByPk($eventData->episode_id);
+
+		$latestData = $this->findAllBySql("
+						SELECT eob.*, '".$values[0]."' AS event_id FROM et_ophtroperationnote_biometry eob
+										WHERE eob.patient_id=".$episodeData->patient_id."
+										AND eob.last_modified_date <= '".$eventData->last_modified_date."'
+										ORDER BY eob.last_modified_date
+										DESC LIMIT 1; ");
+		return $latestData;
+	}
+
+	// because we are working with a view here we do not need to save
+	function __set ($name, $value){
+		return true;
+	}
+
+	function save( $validation = false, $attributes = NULL){
+		return true;
+	}
+}
```

Comment:
These two functions need to have scope declared: `Declaration of Element_OphTrOperationnote_Biometry::save() should be compatible with that of BaseActiveRecordVersioned::save()`